ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -697,22 +697,30 @@ loadGhcSession ghcSessionDepsConfig = do
                 Nothing -> LBS.toStrict $ B.encode (hash (snd val))
         return (Just cutoffHash, val)
 
-    defineNoDiagnostics $ \GhcSessionDeps file -> do
-        env <- use_ GhcSession file
-        ghcSessionDepsDefinition ghcSessionDepsConfig env file
+    defineEarlyCutoff $ RuleWithCustomNewnessCheck (<=) $
+    -- This rule uses a custom newness check that relies on the derived 'Ord' instance:
```

Comment:
But... there's no `Ord` instance for `GhcSessionDeps`? Am I missing something?

Also, this is cute, but maybe it would be safer to just write a trivial but explicit function...

Reply:
Hmm, this is actually quite tricky, so I'm going to retract it and leave it for future work. The `Ord` instance is actually over the `Bytestring` value returned for early cutoff, which in this rule is `Nothing`, so this is absolutely clowny